Test/Resolve-WhiskeyTaskPathParameter.Tests.ps1

```diff
@@ -0,0 +1,692 @@
+        }
+        Init
+        WhenRunningTask 'Task' -Parameter @{ 'Path' = '../YOLO.yml' } -ErrorAction SilentlyContinue
+        ThenTaskNotCalled
+        ThenThrewException 'outside\ the\ build\ root'
+    }
+}
+
+Describe ('Resolve-WhiskeyTaskPathParameter.when path must exist and does') {
+    It ('should pass full path to file') {
+        function global:Task
+        {
+            [Whiskey.Task('Task')]
+            param(
+                [Whiskey.Tasks.ValidatePath(Mandatory,PathType='File')]
+                [string]$Path
+            )
+            $global:taskCalled = $true
+            $global:taskParameters = $PSBoundParameters
+        }
+        Init
+        GivenFile 'abc.yml'
+        WhenRunningTask 'Task' -Parameter @{ 'Path' = 'abc.yml' }
+        ThenTaskCalled -WithParameter @{ 'Path' = (Join-Path -Path $TestDrive.FullName -Childpath 'abc.yml' ) }
+        ThenPipelineSucceeded
+    }
+}
+
+Describe ('Resolve-WhiskeyTaskPathParameter.when path must exist and does not') {
+    It ('should fail') {
+        function global:Task
+        {
+            [Whiskey.Task('Task')]
+            param(
+                [Whiskey.Tasks.ValidatePath(Mandatory,PathType='File')]
+                [string]$Path
+            )
+            $global:taskCalled = $true
+            $global:taskParameters = $PSBoundParameters
+        }
+        Init
+        WhenRunningTask 'Task' -Parameter @{ 'Path' = 'abc.yml' } -ErrorAction SilentlyContinue
+        ThenTaskNotCalled
+        ThenThrewException -Pattern 'does\ not\ exist'
+    }
+}
+
+Describe ('Resolve-WhiskeyTaskPathParameter.when path is absolute') {
+    It ('should succeed') {
+        function global:Task
+        {
+            [Whiskey.Task('Task')]
+            param(
+                [Whiskey.Tasks.ValidatePath(Mandatory,MustExist=$false,PathType='File')]
+                [string]$Path
+            )
+            $global:taskCalled = $true
+            $global:taskParameters = $PSBoundParameters
+        }
+        Init
+        WhenRunningTask 'Task' -Parameter @{ 'Path' = [IO.Path]::GetFullPath((Join-Path -Path $TestDrive.FullName -Childpath 'abc.yml' )) }
+        ThenTaskCalled -WithParameter @{ 'Path' = [IO.Path]::GetFullPath((Join-Path -Path $TestDrive.FullName -Childpath 'abc.yml' )) }
+        ThenPipelineSucceeded
+    }
+}
+
+Describe ('Resolve-WhiskeyTaskPathParameter.when path doesn''t exist but has a wildcard') {
+    It ('should work as expected') {
+        function global:Task
+        {
+            [Whiskey.Task('Task')]
+            param(
+                [Whiskey.Tasks.ValidatePath(Mandatory,MustExist=$false,PathType='File')]
+                [string]$Path
+            )
+            $global:taskCalled = $true
+            $global:taskParameters = $PSBoundParameters
+        }
+        Init
+        WhenRunningTask 'Task' -Parameter @{ 'Path' = 'packages\tool\*.yolo' } -ErrorAction SilentlyContinue
+        if ($PSVersionTable.PSVersion.Major -lt 6)

```

Comment:
If a path contains wildcard characters that don't resolve to anything, don't pass anything to the task.